### PR TITLE
DTD-4358 Cucumber to scalaTest - Shift left analysis - IFS Repo (1)

### PR DESCRIPTION
**Pull request by himeshhp8**
Removed ATs which are covered in UT/IT
[Spreadsheet](https://docs.google.com/spreadsheets/d/1NqQT4u2fMNkjTmAB76gCiBo7FDOCYpLY58eVKqK5yU0/edit?pli=1&gid=1157171139#gid=1157171139) to verify analysis. ATs verified in this ticket are only with the yellow background.
Leave the commented AT as is in InterestRateChanges EdgeCases.feature. A ticket will be raised to investigate it as it has been commented out since 2021.

### CODE REVIEW

File: src/test/resources/features/ifs/DebtCalculationValidation.feature [hunk 1]
```diff
@@ -1,270 +0,0 @@
...
-#      | 1.2            | 2021-03-01  | 2021-03-01        | 2021-03-08          | 1525      | 1000     | true            |
-#    And the debt item has no payment history
-#    And no breathing spaces have been applied to the debt item
-#    And no post codes have been provided for the customer
-#    When the debt item is sent to the ifs service
-#    Then the ifs service will respond with
-#      | statusCode | reason                      | message                                                         |
-#      | 400        | Invalid JSON error from IFS | Field at path '/debtItems(0)/originalAmount' missing or invalid |
-
-  Scenario: TPSS MainTrans (1525) debt invalid entry in Date Created - Edge Case
```
Comment (balaganesan) [inline src/test/resources/features/ifs/DebtCalculationValidation.feature:48]: In the tracking sheet status is "Dev required to move to codebase", is there any reason to delete now?

File: src/test/resources/features/ifs/DebtCalculationValidation.feature [hunk 2]
```diff
@@ -1,270 +0,0 @@
...
-      | 500000         |             | 2021-03-01        | 2021-03-08          | 1525      | 1000     | true            |
-    And the debt item has no payment history
-    And no breathing spaces have been applied to the debt item
-    And no post codes have been provided for the customer
-    When the debt item is sent to the ifs service
-    Then the ifs service will respond with
-      | statusCode | reason                      | message                                                      |
-      | 400        | Invalid JSON error from IFS | Field at path '/debtItems(0)/dateCreated' missing or invalid |
-
-  Scenario: TPSS MainTrans (1525) debt invalid Date Created - Edge Case
```
Comment (balaganesan) [inline src/test/resources/features/ifs/DebtCalculationValidation.feature:72]: In the tracking sheet status is "Dev required to move to codebase", is there any reason to delete now?

File: src/test/resources/features/ifs/DebtCalculationValidation.feature [hunk 3]
```diff
@@ -1,270 +0,0 @@
...
-      | 500000         | 2021-04-03  | 2021-03-08          | 1525      | 1000     | true            |
-    And the debt item has no payment history
-    And no breathing spaces have been applied to the debt item
-    And no post codes have been provided for the customer
-    When the debt item is sent to the ifs service
-    Then the ifs service will respond with
-      | statusCode | reason                       | message                                                                   |
-      | 400        | Missing Interest Start Date. | Interest Start Date missing. This is mandatory for interest bearing debts |
-
-  Scenario: TPSS MainTrans (1525) debt empty interestRequestedTo - Edge Case
```
Comment (balaganesan) [inline src/test/resources/features/ifs/DebtCalculationValidation.feature:96]: In the tracking sheet status is "Dev required to move to codebase", is there any reason to delete now?

File: src/test/resources/features/ifs/InterestRateChanges EdgeCases.feature [hunk 4]
```diff
@@ -1,196 +1,118 @@
...
-# Interest requested to (add in old value) 31/03/2021
-# No repayments
-# No suppression
-# No breathing space
-
-
-# DTD-172 - Request interest for Drier case (interest rate changes)
-Feature: Interest Rate Changes - Edge cases
-
-  Scenario: 300 Debt items - Interest rate changes from 3.0% to 3.25%
```
Comment (balaganesan) [inline src/test/resources/features/ifs/InterestRateChanges EdgeCases.feature:16]: Looks this is business scenario, please don't remove

File: src/test/resources/features/ifs/InterestRateChanges EdgeCases.feature [hunk 5]
```diff
@@ -1,196 +1,118 @@
...
-    And no post codes have been provided for the customer
-    When the debt items is sent to the ifs service
-    Then the ifs service wilL return a total debts summary of
-      | combinedDailyAccrual | interestDueCallTotal | unpaidAmountTotal | amountIntTotal | amountOnIntDueTotal |
-      | 13200                | 3795900              | 150000000         | 153795900      | 150000000           |
-    And the 300th debt summary will contain
-      | interestBearing | numberChargeableDays | interestDueDailyAccrual | interestDueDutyTotal | unpaidAmountDuty |
-      | true            | 302                  | 44                      | 12653                | 500000           |
-
-  Scenario: 2 Debts - Interest rate changes from 3.25% to 2.75% - leap year - payment is made for 1 debt - Interest rate changes from 2.75% to 2.6%
```
Comment (balaganesan) [inline src/test/resources/features/ifs/InterestRateChanges EdgeCases.feature:28]: Looks this is business scenario, please don't remove

File: src/test/resources/features/ifs/InterestRateChanges EdgeCases.feature [hunk 6]
```diff
@@ -1,196 +1,118 @@
...
-    And the 2nd debt summary will have calculation windows
-      | periodFrom | periodTo   | numberOfDays | interestRate | interestDueDailyAccrual | interestDueWindow | amountOnIntDueWindow |
-      | 2009-01-01 | 2009-01-05 | 4            | 5.5          | 75                      | 301               | 500000               |
-      | 2009-01-06 | 2009-01-26 | 21           | 4.5          | 61                      | 1294              | 500000               |
-      | 2009-01-27 | 2009-03-23 | 56           | 3.5          | 47                      | 2684              | 500000               |
-      | 2009-03-24 | 2009-09-28 | 189          | 2.5          | 34                      | 6472              | 500000               |
-      | 2009-09-29 | 2010-01-01 | 95           | 3.0          | 41                      | 3904              | 500000               |
-
-
-  Scenario:  Interest rate changes from 2.75% to 2.6% - leap year - payment is made for 1 debt on the same day the interest rate changes
```
Comment (balaganesan) [inline src/test/resources/features/ifs/InterestRateChanges EdgeCases.feature:77]: Looks this is business scenario, please don't remove

File: src/test/resources/features/ifs/DebtCalculationValidation.feature [hunk 7]
```diff
@@ -1,37 +1,5 @@
...
-      | -1             | 2021-03-01  | 2021-03-01        | 2021-03-08          | 1525      | 1000     | true            |
-    And the debt item has no payment history
-    And no breathing spaces have been applied to the debt item
-    And no post codes have been provided for the customer
-    When the debt item is sent to the ifs service
-    Then the ifs service will respond with
-      | statusCode | reason                          | message                                                                                                                                                                                  |
-      | 400        | Could not validate Json request | Could not parse body due to requirement failed: Original Amount can be zero or greater, negative values are not accepted; Amount paid in payments cannot be greater than Original Amount |
-
-  Scenario: TPSS MainTrans (1525) debt Amount non integer - Edge Case
```
Comment (balaganesan) [inline src/test/resources/features/ifs/DebtCalculationValidation.feature:22]: The status in the spreadsheet is 'Dev required to move to codebase' but it's deleted already. Don't we need to keep this until unit test write.

File: src/test/resources/features/ifs/InterestRateChanges.feature [hunk 8]
```diff
@@ -1,202 +1,183 @@
...
-      | combinedDailyAccrual | interestDueCallTotal | unpaidAmountTotal | amountIntTotal | amountOnIntDueTotal |
-      | 41                   | 13363                | 500000            | 513363         | 500000              |
-    And the 1st debt summary will contain
-      | interestBearing | interestDueDailyAccrual | interestDueDutyTotal | unpaidAmountDuty | totalAmountIntDuty | numberChargeableDays | amountOnIntDueDuty | interestOnlyIndicator |
-      | true            | 41                      | 13363                | 500000           | 513363             | 366                  | 500000             | false                 |
-    And the 1st debt summary will have calculation windows
-      | periodFrom | periodTo   | numberOfDays | interestRate | interestDueDailyAccrual | interestDueWindow | amountOnIntDueWindow | unpaidAmountWindow |
-      | 2021-04-03 | 2022-01-06 | 278          | 2.6          | 35                      | 9901              | 500000               | 509901             |
-      | 2022-01-07 | 2022-02-20 | 45           | 2.75         | 37                      | 1695              | 500000               | 501695             |
+Feature: Interest Rate Changes
```
Comment (balaganesan) [inline src/test/resources/features/ifs/InterestRateChanges.feature:1]: why this feature file added again ? just in the previously it was removed. I can see again this.

File: src/test/resources/features/ifs/InterestRateChanges EdgeCases.feature [hunk 9]
```diff
@@ -1,196 +1,118 @@
...
+# subTrans (CHARGE_TYPE == Duty Repaid in Error - National Insurance) = 1000, 1090
+# interestStartDate (Interest start date = Date Amount) 01/02/2020
+# interestRateDate (New data item) = 07/04/2020
+# Interest requested to (add in old value) 31/03/2021
+# No repayments
+# No suppression
+# No breathing space
+
+
+# DTD-172 - Request interest for Drier case (interest rate changes)
```
Comment (balaganesan) [inline src/test/resources/features/ifs/InterestRateChanges EdgeCases.feature:13]: looks this is a new addition, just previously added comment for few scenarios. Looks its repeating again. Please check this.

File: src/test/resources/features/ifs/DebtCalculationValidation.feature (hunk 7 above)
Comment (himeshhp8) [reply src/test/resources/features/ifs/DebtCalculationValidation.feature:22]: Image
It says Removal in progress so removed

File: src/test/resources/features/ifs/DebtCalculationValidation.feature (hunk 3 above)
Comment (himeshhp8) [reply src/test/resources/features/ifs/DebtCalculationValidation.feature:96]: Image
That test is kept.

File: src/test/resources/features/ifs/InterestRateChanges.feature (hunk 8 above)
Comment (himeshhp8) [reply src/test/resources/features/ifs/InterestRateChanges.feature:1]: I don't know how this happened. I only removed the AT marked for removal

File: src/test/resources/features/ifs/InterestRateChanges EdgeCases.feature (hunk 4 above)
Comment (himeshhp8) [reply src/test/resources/features/ifs/InterestRateChanges EdgeCases.feature:16]: Image
It is already kept

File: src/test/resources/features/ifs/InterestRateChanges EdgeCases.feature (hunk 5 above)
Comment (himeshhp8) [reply src/test/resources/features/ifs/InterestRateChanges EdgeCases.feature:28]: Image
It is already there

File: src/test/resources/features/ifs/InterestRateChanges EdgeCases.feature (hunk 6 above)
Comment (himeshhp8) [reply src/test/resources/features/ifs/InterestRateChanges EdgeCases.feature:77]: Image
It is already there

File: src/test/resources/features/ifs/InterestRateChanges EdgeCases.feature (hunk 9 above)
Comment (himeshhp8) [reply src/test/resources/features/ifs/InterestRateChanges EdgeCases.feature:13]: Looks good to me

File: src/test/resources/features/ifs/DebtCalculationValidation.feature (hunk 1 above)
Comment (himeshhp8) [reply src/test/resources/features/ifs/DebtCalculationValidation.feature:48]: Added back

File: src/test/resources/features/ifs/DebtCalculationValidation.feature (hunk 2 above)
Comment (himeshhp8) [reply src/test/resources/features/ifs/DebtCalculationValidation.feature:72]: Added back

File: src/test/resources/features/ifs/InterestRateChanges.feature (hunk 8 above)
Comment (balaganesan) [reply src/test/resources/features/ifs/InterestRateChanges.feature:1]: verified in the IntelliJ looks ok.

**Review by balaganesan — APPROVED** [review]
